js/main.js

```diff
@@ -317,26 +327,29 @@ function cityIdToLabel(s) {
 /*
  * Set the current city.
  *
- * `city` is the city ID.
+ * `cityID` is the new city's ID. If `newHistoryState` is true then a new
+ * entry in the browser's history is created for the change. Otherwise the
+ * current history entry is replaced.
  */
-function setCity(city) {
-    var filename = 'cities/' + city + '.json';
+function setCity(cityID, newHistoryState) {
```

Comment:
Sorry, but I don't see how adding a new function that simply curries the existing function improves readability of the code. We're not talking about a function with 10+ parameters here. I'd rather have a single function with 5 well documented parameters than 5 functions with one parameter each. Plus I'd also have to document (and ideally test) the curried versions. I'd go with DRY + good docs.

Take `history.pushState` as an example: It has three parameters (two of which are mandatory) and it is [well documented](https://www.w3.org/TR/html5/browsers.html#dom-history-pushstate). I think that's much better than having `pushStateDataKeepTextKeepUrl`, `pushStateTextKeepDataKeepUrl`, `pushStateDataAndTextKeepUrl`, etc.

If you're ever running into the situation that a function's interface gets too complex because of many optional parameters then the proper JavaScript pattern would be to [pass the optional parameters via a map](http://www.openjs.com/articles/optional_function_arguments.php). However, with just two parameters that doesn't make sense in our situation.